File: products/src/main/java/com/github/products/services/IProductService.java
package com.github.products.services;

import com.github.products.entity.EntityStatus;
import com.github.products.entity.Product;

import java.util.List;

public interface IProductService {

    List<Product> searchProduct(String name, String description);

    Product create(Product p);

    Product readById(Long id);

    Product getById(Long id);

    List<Product> readAllByStatusOff();

    List<Product> readAllByIds(List<Long> ids);

    void update(Product p);

    void updateProduct(Product p);

    void updateStatus(EntityStatus status, Long id);

    void updateBoughtCountPlus(Long id);

    void updateBoughtCountMinus(Long id);

}

File: server-mocks/src/main/kotlin/com/github/server/mocks/services/IProductService.kt
package com.github.server.mocks.services

import com.github.server.mocks.payload.Product
import com.github.server.mocks.payload.ProductStatus
import kotlin.reflect.jvm.internal.impl.load.kotlin.JvmType

interface IProductService {

    fun create(product: Product): Product

    fun readById(id: Long): Product

    fun readByIds(ids: List<Long>): List<Product>

    fun readAll(): List<Product>

    fun update(product: Product)

    fun updateStatus(id: Long, status: ProductStatus)

}